framework/src/meshgenerators/StitchedMeshGenerator.C

```diff
@@ -129,6 +134,32 @@ StitchedMeshGenerator::generate()
 
     const bool use_binary_search = (_algorithm == "BINARY");
 
+    // Avoid chaotic boundary merging due to overlapping ids in meshes by simply renumbering the
+    // boundaries in the meshes we are going to stitch onto the base mesh
+    // This is only done if there is an overlap
+    if (_prevent_boundary_ids_overlap)
+    {
+      const auto & base_mesh_bids = mesh->get_boundary_info().get_global_boundary_ids();
+      const auto stitched_mesh_bids = meshes[i]->get_boundary_info().get_global_boundary_ids();
+      // Check for an overlap
+      bool overlap_found = false;
+      for (const auto & bid : stitched_mesh_bids)
+        if (base_mesh_bids.find(bid) != base_mesh_bids.end())
+          overlap_found = true;
+
+      if (overlap_found)
+      {
+        const auto max_boundary_id = *base_mesh_bids.rbegin();
+        int new_index = 0;
+        for (const auto bid : stitched_mesh_bids)
+        {
+          meshes[i]->get_boundary_info().renumber_id(bid, max_boundary_id + (new_index++));
+          if (bid == second)
+            second = max_boundary_id + new_index - 1;
```

Comment:
Why this special treatment?

Reply:
if you rename the bid without adapting second, stitching cant find second in the mesh?

Reply:
Oh I understand now. The subtraction with `- 1` threw me a bit. Could you make a new local variable like `new_bid` that makes it clear the argument to `renumber_id` and the assignment to `second` are the same?